### PR TITLE
Update README and code to ES6, and latest Ember

### PR DESCRIPTION
**Pull request by rwwagner90**
Had some issues because the documentation did not indicate the new syntax, so I decided to update it.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -37,24 +37,36 @@ Will result into this HTML (open graph metas are using property automatically) :
 
 Title & metas are set only on `didTransition` in case the route would be canceled.
 
+**Note**: og tags will not correctly work without using FastBoot or some sort of server side rendering.
```
Comment (j15e) [inline README.md]: for bots not fully executing JavaScript (Google does).

https://webmasters.googleblog.com/2015/10/deprecating-our-ajax-crawling-scheme.html

Comment (rwwagner90) [inline README.md]: @j15e I was talking about what you would use og tags for, like posting to facebook, twitter, etc. I was unaware Google even look at the og tags, thought they had all their own metas they looked at.

**Comment by j15e** [comment]
Thanks that is a nice 2016 update. The new "syntax" is actually Javascript ES6 which requires transcompilation for all browsers support. It is now built in EmberJS / babel so not an issue.

File: README.md (hunk 1 above)
Comment (j15e) [inline README.md]: You are right, I don't link Google look after the Open Graph tags.

File: README.md [hunk 2]
```diff
@@ -15,7 +15,7 @@ Set desired metas in your controller setup :
 
 ```js
 export default Ember.Route.extend({
-  setupController: function (controller, model) {
+  afterModel() {
```
Comment (j15e) [inline README.md]: Any reason to prefer `afterModel` over `setupController`?

**Comment by rwwagner90** [comment]
Babel transpilation has been built into Ember for a couple years now, definitely no browser issues 😄

File: README.md (hunk 2 above)
Comment (rwwagner90) [inline README.md]: `setupController` is generally discouraged these days. I have seen several people having issues with it in the Ember slack channel. It also stops my model from being populated with data if I used `setupController`, so I think `afterModel` is the correct choice for the current Ember standards. Also, controllers are supposed to be going away at some point.

**Comment by j15e** [comment]
Yeah sorry I am living in the past of ember, we created our major ember app 2 years ago so I still got some bad reflex (maybe soon enough every browser will support it too eheh).

**Comment by rwwagner90** [comment]
Yeah, Ember has changed a lot in 2 years. The day that all browsers support ES2015+ will be great 😃

File: README.md (hunk 1 above)
Comment (j15e) [inline README.md]: I would add a link to Fastboot & say Open Graph (og) to be clear.

```md
**Note**: Open Graph (og) tags will not correctly work without using [FastBoot](https://www.ember-fastboot.com/) or some sort of server side rendering.
```

I can do it after merging

Comment (rwwagner90) [inline README.md]: I can update the PR, just give me a minute

**Comment by rwwagner90** [comment]
Updated the README to be more specific about open graph tags

File: README.md (hunk 2 above)
Comment (j15e) [inline README.md]: Ok yeah, thanks for the explanation.

File: README.md [hunk 3]
```diff
@@ -37,24 +37,36 @@ Will result into this HTML (open graph metas are using property automatically) :
 
 Title & metas are set only on `didTransition` in case the route would be canceled.
 
+**Note**: Open Graph tags, for sharing on Facebook or Twitter, will not correctly work without using [FastBoot](https://www.ember-fastboot.com/) or some sort of server side rendering. All Google tags and data will work fine though because Google executes the JS.
```
Comment (j15e) [inline README.md]: Perfect like that, its very clear

**Comment by j15e** [comment]
I see two deprecation warnings in the tests (maybe they were already present before your changes), but do you think we can fix them at the same time?

> DEPRECATION: The `initialize` method for Application initializer \\\'meta\\\' should take only one argument - `App`, an instance of an `Application`. [deprecation id: ember-application.app-initializer-initialize-arguments] See http://emberjs.com/deprecations/v2.x/#toc_initializer-arity for more details.

**Comment by rwwagner90** [comment]
The deprecations are because of your attempt at supporting old and new initializer types. `initialize: function(app, oldApp) {` We would need to remove `oldApp` there.

With your blessing, I would like to do a separate PR after this and update to the latest Ember-CLI and get rid of the checks to support old 1.13 Ember. I'd recommend creating a legacy-1.13 branch with the current code, for anyone who may still need that support.

Thoughts @j15e?

**Comment by j15e** [comment]
Yeah seems legit! Will create this branch after this merge / before dropping 1.13.

**Comment by j15e** [comment]
I think I will release a last version with support for 1.13 too, with this update

**Comment by rwwagner90** [comment]
Releasing once more sounds like a good idea, then we should update the README to say, if on 1.13, use that specific version, otherwise install latest.

**Pull request merged**